bzzeth/bzzeth.go

```diff
@@ -266,10 +293,22 @@ func (b *BzzEth) deliverAndStoreAll(ctx context.Context, req *request, headers [
 
 	// wait for all validations to get over and close the channels
 	err := wg.Wait()
+
+	// Store all the valid header chunks in one shot
+	storeErr := b.storeChunks(ctx, chunks)
+	if storeErr != nil {
```

Comment:
oops... i should have returned the storeErr if it is not null..
Fixed it.

Reply:
Actually, the problem is also that the error from wg.Wait() is handled after this code block, but should be before.

Reply:
@janos This is intentional..we want to store even if there is any validation error since this is validated in a batch.. There can be situations where few headers are invalid and few are valid... in that case we have to store the valid ones... but send back the validation error so that peer can be disconnected.

Reply:
Thanks for the explanation. That was not so clear to me. Maybe a comment about that reason would be nice to have in the codebase.

Reply:
The errgroup returns on the first error, so not sure how much it is worth